### PR TITLE
Use letsencrypt for making certs.

### PR DESCRIPTION
**Pull request by mcav**
Tested by running on the hosted service box.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -48,7 +49,7 @@ class SpaceKitService {
       this.tlsServer = CreateTlsProxyServer(this.handleTlsConnection.bind(this));
       this.tlsServer.listen(443);
       // Listen for any incoming Net connections.
-      this.plainServer = CreateNetProxyServer(this.handleNetConnection.bind(this));
+      this.plainServer = CreateNetProxyServer(this.handlePlaintextConnection.bind(this));
```
Comment (jedireza) [inline lib/index.js]: `handlePlainTextConnection`

File: lib/index.js [hunk 2]
```diff
@@ -60,8 +61,11 @@ class SpaceKitService {
     // established connections from a TLS proxy handler.
     if (config.apiKey) {
       this.apiServer = Https.createServer({
-        key: Fs.readFileSync(config.apiKey),
-        cert: Fs.readFileSync(config.apiCert)
+        SNICallback: (serverName, cb) => {
```
Comment (jedireza) [inline lib/index.js:64]: `SniCallback`

File: lib/index.js [hunk 3]
```diff
@@ -84,8 +88,9 @@ class SpaceKitService {
     // a TLS proxy handler.
     if (config.webKey) {
       this.webServer = Https.createServer({
-        key: Fs.readFileSync(config.webKey),
-        cert: Fs.readFileSync(config.webCert)
+        SNICallback: (serverName, cb) => {
```
Comment (jedireza) [inline lib/index.js:91]: `SniCallback`

Comment (mcav) [inline lib/index.js:91]: That's part of the node api, we can't change that one

File: lib/index.js [hunk 4]
```diff
@@ -148,31 +160,29 @@ class SpaceKitService {
    * relay, so that users can run providers like Let's Encrypt themselves to
    * receive certificates.
    */
-  handleNetConnection (socket, hostname, path) {
+  handlePlaintextConnection (socket, hostname, path) {
```
Comment (jedireza) [inline lib/index.js]: `handlePlainTextConnection`

**Comment by jedireza** [comment]
👍

File: lib/index.js (hunk 3 above)
Comment (jedireza) [inline lib/index.js:91]: Gotcha

**Pull request merged**